libs/newsletter-workflow/src/lib/calculateFieldsFromName.ts

```diff
@@ -0,0 +1,34 @@
+import { deriveNewsletterFieldsFromName } from '@newsletters-nx/newsletters-data-client';
+import type { WizardFormData } from '@newsletters-nx/state-machine';
+
+export const calculateFieldsFromName = (
+	formData: WizardFormData,
+): WizardFormData => {
+	const derivedFields = deriveNewsletterFieldsFromName(formData.name as string);
```

Comment:
Great stuff! I would suggest refactoring this function slightly so :
- it takes the name as argument rather than casting it  from the `WizardFormData.name` - it would be a slightly more robust function that way ;and 
- it returns an object (`DerivedDraftFields`?) typed to only contain the derived fields instead of a new `WizardFormData` with the existing inputs include. Just to make the flow a bit clearer. We should still be able to spread that object into the `DraftWithoutId`  - I'll put another comment below on the other file.

maybe something like:

```typescript
export const calculateFieldsFromName = (
	formName: string,
): DerivedDraftFields => {
	const derivedFields = deriveNewsletterFieldsFromName(formData.name as string);
```